sql/core/src/test/scala/org/apache/spark/sql/MathFunctionsSuite.scala

```diff
@@ -233,6 +233,18 @@ class MathFunctionsSuite extends QueryTest with SharedSQLContext {
     )
   }
 
+  test("round/bround with data frame from a local Seq of Product") {
+    val df = spark.createDataFrame(Seq(NumericRow(BigDecimal("5.9"))))
```

Comment:
Actually, the problem occurs only when creating data frame from `Product`. Unable to reproduce the issue with `Seq(BigDecimal("5.9")).toDF("value")`